### PR TITLE
Export ProgressBar to allow it to be used

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
Export `ProgressBar` component as `__experimentalProgressBar`

## Why?
To allow it to be used.

## Testing Instructions
In a react component add:

```
import { _experimentalProgressBar as ProgressBar } from '@wordpress/components';


function TestComponent () {
    return (
      <div> HI! <ProgressBar /> </div>
    )
}

```

### CODE REVIEW

**Review by alexstine — COMMENTED** [review]
@matiasbenedetto This will be good to go once it has a changelog entry.

**Comment by matiasbenedetto** [comment]
@alexstine thanks for the review!
I just added the changelog entry :)

**Review by alexstine — APPROVED** [review]
👍

**Comment by ciampo** [comment]
Hey folks, I have a couple of comments:

- Let's not export components anymore with the `__experimental` prefix, since it has been determined that code merged into WordPress core can't be assumed as experimental, regardless of naming conventions. A couple of alternatives:
  - If we feel that the component is still experimental, we should instead use the lock/unlock APIs ([example](https://github.com/WordPress/gutenberg/blob/162eb4b561b7f5ccc6da6f8c5e32ebd722f779f5/packages/components/src/private-apis.ts))
  - Otherwise, if the component is mature enough, we could export it without the additional `__experimental` prefix
- the CHANGELOG entry should always be under a sub-section (in this case, `### Experimental` if we keep the component as an experiment, or otherwise `### New feature`)

Hey @tyxla , since you worked on the component, do you have a feeling on whether we should consider it as experimental or not?

**Comment by ciampo** [comment]
Nevermind my previous comment, I have just noticed that `ProgressBar` is indeed [already exported via private APIs ](https://github.com/WordPress/gutenberg/blob/162eb4b561b7f5ccc6da6f8c5e32ebd722f779f5/packages/components/src/private-apis.ts#L50) and [used in other parts of the editor](https://github.com/WordPress/gutenberg/blob/2af5a244678f1e191b43ba8d03530ea5b6c27690/packages/edit-site/src/components/canvas-loader/index.js#L15).

I will therefore open a new PR reverting the changes from the current one.

Please let's be careful in the future when exposing new APIs from the package (and approving such changes); given the WordPress strict backwards compatibility policy, it's really hard (if not impossible) to undo such changes.

In case the current way of consuming the `ProgressBar` was somehow not enough, let's discuss in a separate issue.

**Comment by tyxla** [comment]
> Nevermind my previous comment, I have just noticed that `ProgressBar` is indeed already exported via private APIs (lock/unlock).
> 
> I will therefore open a new PR reverting the changes from the current one.
> 
> Please let's be careful in the future when exposing new APIs from the package (and approving such changes); given the WordPress strict backwards compatibility policy, it's really hard (if not impossible) to undo such changes.

Spot on. We're indeed already exporting it by using the recommended `lock/unlock` mechanism for private APIs, and that's the recommended way to consume that component.

I wasn't pinged for reviewing this PR, otherwise, I would have caught it. This is also how we're using the component in the site editor loading experience right now.

To your question @ciampo, I'd prefer to keep it private for the time being, since it's a rather new and simplistic component and I'm not sure if we're going to keep the same API. IMO we could potentially stabilize it after it lands in the next WP release.

**Comment by tyxla** [comment]
Sharing again for posterity, this is the recommended way to consume the `ProgressBar` component right now:

https://github.com/WordPress/gutenberg/blob/2af5a244678f1e191b43ba8d03530ea5b6c27690/packages/edit-site/src/components/canvas-loader/index.js#L15

**Comment by matiasbenedetto** [comment]
Thanks for the fix

**Comment by alexstine** [comment]
@ciampo @tyxla Is there any documentation around how we handle private APIs? I would like to learn more about how components/packages are exported in the project.

**Comment by ciampo** [comment]
> @ciampo @tyxla Is there any documentation around how we handle private APIs? I would like to learn more about how components/packages are exported in the project.

@alexstine this is not a components package specific feature, but a Gutenberg-wide one. You can read more of these APIs at this page: https://developer.wordpress.org/block-editor/reference-guides/packages/packages-private-apis/

**Pull request merged**
